src/Analysis/Ast/Impl/Specializations/Typing/Types/GenericTypeParameter.cs

```diff
@@ -33,27 +35,67 @@ public GenericTypeParameter(string name, IPythonModule declaringModule, IReadOnl
         public override PythonMemberType MemberType => PythonMemberType.Generic;
         public override bool IsSpecialized => true;
 
-
-        public static IPythonType FromTypeVar(IArgumentSet argSet, IPythonModule declaringModule, IndexSpan location = default) {
+        private static bool TypeVarArgumentsValid(IArgumentSet argSet) {
             var args = argSet.Values<IMember>();
+            var eval = argSet.Eval;
+            var callExpression = argSet.CallExpression;
+            var callLocation = callExpression?.GetLocation(eval?.Module);
+
             if (args.Count == 0) {
-                // TODO: report that at least one argument is required.
-                return declaringModule.Interpreter.UnknownType;
+                eval?.ReportDiagnostics(
+                    eval.Module?.Uri,
+                    new DiagnosticsEntry(Resources.TypeVarMissingFirstArgument,
+                    callLocation?.Span ?? default,
+                    Diagnostics.ErrorCodes.TypeVarArguments,
+                    Severity.Error, DiagnosticSource.Analysis)
+                );
+
+                return false;
             }
 
             var name = (args[0] as IPythonConstant)?.GetString();
             if (string.IsNullOrEmpty(name)) {
-                // TODO: report that type name is not a string.
-                return declaringModule.Interpreter.UnknownType;
+                var firstArgLocation = callExpression?.Args[0]?.GetLocation(eval?.Module);
+                eval?.ReportDiagnostics(
+                    eval.Module?.Uri,
+                    new DiagnosticsEntry(Resources.TypeVarFirstArgumentNotString,
+                    firstArgLocation?.Span ?? default,
+                    Diagnostics.ErrorCodes.TypeVarArguments,
+                    Severity.Warning, DiagnosticSource.Analysis)
+                );
+
+                return false;
+            }
+
+            if (args.Skip(1).Count() == 1) {
```

Comment:
Since `ArgumentSet.Arguments` is `IReadOnlyList` count is known so you can simply do `args.Count-1 == 1`

Reply:
Thanks.